internal/store/builder.go

```diff
@@ -461,7 +472,13 @@ func (b *Builder) buildStores(
 			familyHeaders,
 			composedMetricGenFuncs,
 		)
-		listWatcher := listWatchFunc(b.kubeClient, v1.NamespaceAll, b.namespaceFilter)
+		merged, err := b.MergeFieldSelector(b.namespaceFilter, b.nodenameFilter)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to merge fieldSelector %s and %s", b.namespaceFilter, b.nodenameFilter))
+		}
+		klog.Infof("FieldSelector is used ", merged)
+		listWatcher := listWatchFunc(b.kubeClient, v1.NamespaceAll, merged)
+		b.startReflector(expectedType, store, listWatcher, useAPIServerCache)
```

Comment:
Not sure if I am reading the diff correctly, but does this start the same reflector twice?

Reply:
Yes, you're right. I need to remove this `b.startReflector(expectedType, store, listWatcher, useAPIServerCache)`.  Thank you!